Use the first option value presentation as the variant title

thisiscab opened the pull request:
In Shopify we were representing the variant title with the following string
`price: 10$` for a gift card

Now it will only show:
`10$`

thisiscab committed:
Use the first option value presentation as the variant title

The previous way of doing it would display the following string:
price: 10$

Now it will only show:
10$

DanielWright commented:
What do you think of introducing a presenter here, to less tightly couple the Shopify product title to the Spree model layer? Or d'you think it's overkill at this point?

thisiscab committed:
Use the presenter to abstract some of the logic

It's not the goal of the converter to know how to have the logic to know
what's title of the variant. The converter should only have one goal,
which is to transform the variant into a hash.

I've moved the logic into the presentation layer, of which I had
forgotten.

thisiscab commented:
@DanielWright 💃 

The presenter part of it was already existing and the converter already gets the presented_object and not the object itself. I had just forgotten that we have had that logic laying around. 

The change was quite trivial to do, thanks for the reminder! 🗡

DanielWright reviewed (approved):
Nice! :shipit:
